mobile/src/Student/screens/PhaseResult/index.js

```diff
@@ -7,167 +7,104 @@ import TeamFooter from '../../../components/TeamFooter'
-        } else {
-            setSelectedAnswer({
-                id: Number.MIN_VALUE,
-                text: Number.MIN_VALUE,
-            })
-        }
-
-        setSelectedTrickAnswer(ModelHelper.getSelectedTrickAnswer(updatedCurTeam, currentQuestion.id))
-        setLoadedData(true)
-        setTeamInfo(updatedCurTeam, updatedCurTeam.teamMembers[0])
-      }, [gameSession?.currentState])
-    )
-
-    const setAnswer = (answer) => {
-        setIsAnswerCorrect(answer.text === ModelHelper.getCorrectAnswer(currentQuestion).text)
-        if (answer.isTrickAnswer) {
-            setSelectedTrickAnswer(answer)
-        } else {
-            setSelectedAnswer(answer)
-        }
-    }
-
-    const alphabets = ["A", "B", "C", "D"]
-    const getAnswerMode = (choiceText) => {
-        switch (phaseNo) {
-            case 1:
-                if (correctAnswer.text === choiceText) {
-                    return AnswerMode.RightAnswer
-                } else if (choiceText === selectedAnswer.text) {
-                    return AnswerMode.UserAnswer
-                }
-                return AnswerMode.ShowEmptyRightIcon
-            case 2:
-                if (correctAnswer.text === choiceText) {
-                    return AnswerMode.Disabled
-                } else if (selectedTrickAnswer.text === choiceText) {
-                    return AnswerMode.PopularTrickAnswer
-                }
-                break
+        const findSelectedAnswer = (teamAnswers) => {
+          return teamAnswers.find(teamAnswer => ((phaseNo == 1) ? (teamAnswer.isChosen === true) : (teamAnswer.isTrickAnswer === true)))
         }
-    }
-
-    const calculatePercentage = (answer) => {
-        if (isNullOrUndefined(answer)) {
-            setPhase2Score(0)
-            return 0
+        const selectedAnswer = findSelectedAnswer(ModelHelper.getBasicTeamMemberAnswersToQuestionId(team, currentQuestion.id))
+
+        const getP1AnswerMode = (choiceText) =>{
+          if (choiceText === correctAnswer.text) 
+            return AnswerMode.RightAnswer
+          else if (choiceText === selectedAnswer.text) 
+            return AnswerMode.UserAnswer
+          else
+            return AnswerMode.ShowEmptyRightIcon
+        } 
+    
+        const getP2AnswerMode = (choiceText) => {
+          if (choiceText === correctAnswer.text) 
+            return AnswerMode.Disabled
+          else if (choiceText === selectedAnswer.text) 
+            return AnswerMode.PopularTrickAnswer
         }
-
-        let percentage = ModelHelper.calculateBasicModeWrongAnswerScore(gameSession, answer.text, currentQuestion.id)
-
-        if (selectedTrickAnswer?.text === answer.text) {
-            setPhase2Score(percentage)
-        } else {
-            setPhase2Score(0)
+    
+        const getAnswer = ({item, index}) => {
+          return <Answer
+            icon={teamAvatar.smallSrc}
+            text={`${alphabets[index]}. ${item.text}`}
```

Comment:
Wondering if we should add a `title` or `orderedText` to question answers and do this logic there. Definitely a work for a follow up PR, but that helps with not duplicating this logic on Host and mobile and make sure both show the same order and alphabet letter

Reply:
Yeah, that makes a lot of sense. I've created https://github.com/rightoneducation/righton-app/issues/567 to track this issue.